### PR TITLE
Authorization support

### PR DESCRIPTION
**Pull request by p53**
created authorization plugin support, authorization hook in processor and implemented opa authorization plugin, if somebody questioning why i didn't use schema to decode authorization infromation, it is because it would decode whole request and we don't need it for authorization we only need api key, token/user, topic, src/dst ip, we want to be fast...

### CODE REVIEW

**Comment by everesio** [comment]
Thank you for your contribution. I will need some time to review this PR.

**Comment by p53** [comment]
@everesio i made guide to README, i can make it even more detailed to be more easy to try

File: README.md [hunk 1]
```diff
@@ -312,6 +312,169 @@ Connect through test HTTP Proxy server using CONNECT method
...
+Deploy kafka with zookeepers
+
+Example docker-compose, this will deploy kafka listening on port 9094, you should give docker-compose up,
+After start, kafka will fail because it doesn't have proper IP in {YOUR_DOCKER_CONTAINER_IP}, you should give
+docker stop your kafka container, docker inspect your kafka container, replace {YOUR_DOCKER_CONTAINER_IP} with
+IP of container and give again docker-compose up:
+
+
+```
+version: '2'
```
Comment (p53) [inline README.md:330]: @everesio here is docker-compose

**Comment by everesio** [comment]
@p53  Could you rebase your PR on current master ?

**Comment by p53** [comment]
> @p53 Could you rebase your PR on current master ?

done

**Comment by p53** [comment]
@everesio done

**Comment by everesio** [comment]
@p53  Your rebase / merge overwrites some changes from master. Could you rework you PR ?

**Comment by p53** [comment]
> @p53 Your rebase / merge overwrites some changes from master. Could you rework you PR ?

@everesio i think touched only code which was related to my pull request to be functional at it should not affect recent changes, but will look at it to verify

**Comment by everesio** [comment]
@p53 I guess that last rebase has overwritten some changes which were present in the master

**Comment by p53** [comment]
> @p53 I guess that last rebase has overwritten some changes which were present in the master

ah ok tls-same flag did think that it was renamed from tls to same and not other way, will fix it

**Comment by p53** [comment]
@everesio - that flag fixed

**Comment by p53** [comment]
@everesio this pull request won't work, it is clashing with https://github.com/grepplabs/kafka-proxy/commit/fec23783a640a1dc4d0feb6aad9b3bfeeea54d65, will fix it

**Comment by p53** [comment]
@everesio updated that clash+fixed some bugs i found in code+improved error reporting, i made smaller pull requests for improvements/bug i found so that they can be merged sooner

**Pull request closed without merging**